webapp/src/pages/ChatRoom/components/MessageInput/MessageInput.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useRef, useState } from 'react';
+import { Button, ButtonVariant } from '../../../../shared/components';
+import socket from '../../../../shared/socket';
+import './MessageInput.css';
+
+type MessageInputProps = {
+  to: string;
+};
+
+const MessageInput = ({ to }: MessageInputProps) => {
+  const [value, setValue] = useState('');
+  const divRef = useRef<HTMLDivElement>(null);
+
```

Comment:
Could you explain me why `useRef `and not only `useState`, and what is this `dataset.replicatedValue`?

Reply:
You are right :+1:. We could also use only the `useState`. I've changed it.

The replicated value is a [data attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*), we use it to change the content of the div using CSS.

This component has a textarea, which cannot expand its height, and a div, which can do it based on its content.

They share the same grid position, and the height expands depending on which element takes more space.

Whenever the textarea input changes, we copy the value to the data replicated value attribute of the div and assign the div content to this data attribute in CSS.